generate/generate.go

```diff
@@ -310,22 +322,27 @@ func (gen *generatedStruct) write(w io.Writer) error {
 // Nested fields are handled by recursively calling this function until the leaf field is reached.
 // For all fields the name and jsonType are set, for arrays the array bool is set for true and for JSON objects,
 // the fields map is created and if it exists the requiredFields section populated.
-func addField(fields extractedFields, tree []string, inst jsonschema.Instance) error {
+// fields will be renamed if a matching entry is supplied in the fieldRenameMap
+func addField(fields extractedFields, tree []string, inst jsonschema.Instance, fieldRenameMap map[string]string) error {
 	if len(tree) > 1 {
 		if f, ok := fields[tree[0]]; ok {
-			return addField(f.fields, tree[1:], inst)
+			return addField(f.fields, tree[1:], inst, fieldRenameMap)
 		}
 		f := &extractedField{jsonName: tree[0], jsonType: "object", name: exportedName(tree[0]), fields: make(map[string]*extractedField)}
 		fields[tree[0]] = f
-		if err := addField(f.fields, tree[1:], inst); err != nil {
+		if err := addField(f.fields, tree[1:], inst, fieldRenameMap); err != nil {
 			return fmt.Errorf("failed field %q: %v", tree[0], err)
 		}
 		return nil
 	}
 
+	fieldName := fieldRenameMap[tree[0]]
```

Comment:
is there an instance where tree will have a length of 0?

Reply:
Good question, I'll see if I can demonstrate that.

Reply:
Thanks Tom